nixos/modules/system/activation/bootspec.nix

```diff
@@ -79,9 +75,16 @@ let
 in
 {
   options.boot.bootspec = {
-    enable = lib.mkEnableOption (lib.mdDoc "Enable generation of RFC-0125 bootspec in $system/bootspec, e.g. /run/current-system/bootspec");
+    enable = lib.mkEnableOption (lib.mdDoc "the generation of RFC-0125 bootspec in $system/boot.json, e.g. /run/current-system/boot.json")
+      // { default = true; internal = true; };
+    enableValidation = lib.mkEnableOption (lib.mdDoc ''the validation of bootspec documents for each build.
+      This will introduce Go in the build-time closure as we are relying on [Cuelang](https://cuelang.org/) for schema validation.
+      Enable this option if you want to ascertain that your documents are correct.
+      ''
+    );
 
     extensions = lib.mkOption {
+      # FIXME(RaitoBezarius): this is not enough to validate: extensions."osRelease" = drv; those are picked up by cue validation.
```

Comment:
Unknown to me, I'm not enough knowledgeable in the NixOS module internals, I will invoke @ncfavier who might have an idea :D

Reply:
IMHO I don't think extensions are something we can or should try to validate. Extensions were purposely left unspecified in the RFC (with the exception of `specialisation`s), so I don't think there's any kind of validation we can do here other than make sure it's an attrset. It's up to tools themselves how they want to consume / shape their extensions.

Unless I'm misunderstanding.

Reply:
This is a very basic validation of type: "extensions = { $namespace = { ... } }`, not validating what is in the namespace, but validating that people do use namespaces.

Reply:
The problem is that semantically `extensions."osRelease" = some that can be coerced to attrset;` will be validated just fine and it should not be, because Nix will coerce.

Reply:
AFAIK, namespaces aren't required / enforced by the RFC, just recommended.

Reply:
Hmm, I am going to bed but as I always forget the spec, you might be right! Let's see tomorrow.

Reply:
I will transform this into a NOTE, will use `attrsOf anything`, @cole-h does that sound good to you?

Reply:
Done!

Reply:
(with bump included)